sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py

```diff
@@ -237,3 +252,46 @@ def _get_workspace_location(self) -> str:
         """Get the workspace location TODO[TASK 1260265]: can we cache this
         information and only refresh when the operation_scope is changed?"""
         return self._all_operations.all_operations[AzureMLResourceType.WORKSPACE].get(self._workspace_name).location
+
+    def _validate_component(self, deployment: Deployment) -> Component:
+        """Validates that the value provided is associated to an existing component
+        or otherwise we will try to create an anonymous component that will be use
+        for batch deployment
+        :param deployment: Batch deployment
+        :type deployment: ~azure.ai.ml.entities._deployment.deployment.Deployment
+        :raises ~azure.ai.ml.exceptions.ValidationException: Raised if BatchDeployment cannot be
+            successfully validated. Details will be provided in the error message.
+        :return: Component entity associated to the values provided in yaml file
+        :rtype: ~azure.ai.ml.entities.Component
+        """
+        component = None
+        if isinstance(deployment.job_definition.component, PipelineComponent):
+            component = self._component_operations.create_or_update(
+                name =  deployment.job_definition.component.name,
+                resource_group_name=self._resource_group_name,
+                workspace_name=self._workspace_name,
+                body = deployment.job_definition.component._to_rest_object(),
+                version=deployment.job_definition.component.version,
+                **self._init_kwargs
+            )
+        elif isinstance(deployment.job_definition.component, str):
+            split_name = deployment.job_definition.component.split(':')
+            if len(split_name) == 2:
+                component = self._component_operations.get(
+                    name = split_name[0],
+                    resource_group_name=self._resource_group_name,
+                    workspace_name=self._workspace_name,
+                    version= split_name[1],
+                    **self._init_kwargs
+                )
```

Comment:
Not sure why are we making a rest call to resolve this, we should comply with expanding the short form and not making Get() call. have you considered using this https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/operations/_operation_orchestrator.py#:~:text=def%20resolve_azureml_id(self%2C%20arm_id%3A%20str%20%3D%20None%2C%20**kwargs)%20%2D%3E%20str%3A?

Reply:
Yes, any reason to go against the existing flow? Else let's keep with paradigm that client does light validation (URI formats are reasonable) but let service handle existence checks